astropy/time/core.py

```diff
@@ -1847,7 +1926,13 @@ def sidereal_time(self, kind, longitude=None, model=None):
                                   wrap_angle=180. * u.degree)
 
         gst = self._erfa_sidereal_time(available_models[model.upper()])
-        return Longitude(gst + longitude, u.hourangle)
+
+        if (model.upper() == 'IAU1982') or (model.upper() == 'IAU1994'):
```

Comment:
This can be simplified to `if model.upper() in ('IAU1982', 'IAU1994'):` 

Arguably more consistent with the way things are set up would be to add something to the dictionary defining the scales and functions, but that can be later.